packages/subcommands/src/lib/Subcommand.ts

```diff
@@ -15,10 +15,40 @@ import {
 
 export class Subcommand<PreParseReturn extends Args = Args, O extends Subcommand.Options = Subcommand.Options> extends Command<PreParseReturn, O> {
 	public parsedSubcommandMappings: SubcommandMappingArray;
+	private caseInsensitiveSubCommands = false;
 
 	public constructor(context: PieceContext, options: O) {
 		super(context, options);
 		this.parsedSubcommandMappings = options.subcommands ?? [];
+
+		const clientOptions = this.container.client.options;
+		if (clientOptions.caseInsensitiveCommands) {
+			// Because slash commands must be lowercase anyways, blanketing all commands to lowercase is safe.
+			this.parsedSubcommandMappings.forEach((cmd) => cmd.name.toLowerCase());
+			this.caseInsensitiveSubCommands = true;
+		}
+
+		if (options.generateDashLessAliases) {
+			for (const mapping of this.parsedSubcommandMappings) {
+				if (!('messageRun' in mapping)) {
```

Comment:
```suggestion
				if (!Reflect.has(mapping, 'messageRun') || (Reflect.has(mapping, 'messageRun') && Reflect.has(mapping, 'chatInputRun'))) {
					continue;
				}
```

Whatever we go for, it won't be perfect. But I think the safest way to go about it is to only apply dashless when 
1. The option is enabled
2. The command *only* has `messageRun`
3. The command does NOT have `chatInputRun`